filo-scala/src/main/scala/org.velvia.filo/FiloVector.scala

```diff
@@ -25,18 +28,21 @@ object FiloVector {
    *            position will be restored to its original value, but it may change in the meantime.
    */
   def apply[A](buf: ByteBuffer, emptyLen: Int = 0)(implicit cm: VectorReader[A]): FiloVector[A] = {
-    if (buf == null) return new EmptyFiloVector[A](emptyLen)
-    val origPos = buf.position
-    buf.order(ByteOrder.LITTLE_ENDIAN)
-    val headerBytes = buf.getInt()
-    val vector = majorVectorType(headerBytes) match {
-      case VECTORTYPE_EMPTY =>
-        new EmptyFiloVector[A](emptyVectorLen(headerBytes))
-      case other =>
-        cm.makeVector(buf, headerBytes)
+    // if (buf == null) return new EmptyFiloVector[A](emptyLen)
+    val res = Try{
```

Comment:
I think instead of using Try, a faster and better method would be to use `Option(buf).map(....).getOrElse`